api/s2n.h

```diff
@@ -579,7 +579,7 @@ typedef enum {
     S2N_EXTENSION_EC_POINT_FORMATS = 11,
     S2N_EXTENSION_SIGNATURE_ALGORITHMS = 13,
     S2N_EXTENSION_ALPN = 16,
-    S2N_EXTENSION_CERTIFICATE_TRANSPARENCY = 18,
+    S2N_EXTENSION_SCT_LIST = 18,
```

Comment:
This is a public enum, so this is a breaking change. Any application with "S2N_EXTENSION_CERTIFICATE_TRANSPARENCY" in their code will run into build errors when they upgrade their s2n-tls library.

Reply:
Right, we can have both identifiers initialized to 18 (instead of removing the old one). The old identifier will be for backwards compatibility and the new identifier enables the use of the correct naming convention from now on, in the library.

Reply:
I'm not convinced that the name "S2N_EXTENSION_SCT_LIST" is enough of an improvement over "S2N_EXTENSION_CERTIFICATE_TRANSPARENCY" to justify a duplicate value here. The actual name of the extension is "signed_certificate_timestamp" in the RFC, but the RFC / feature itself is called "Certificate Transparency". Users don't necessarily care that the name doesn't match the RFC or our internal code, as long as they can understand the name and recognize that it provides the feature they need.

Reply:
If we're going to rename it anything, we should match the RFC: S2N_EXTENSION_SIGNED_CERTIFICATE_TIMESTAMP. But even then, unless we have an example of this causing confusion for a customer, I don't think the rename is worth adding a duplicate enum value.

Reply:
I agree. A similar story is happening with the [OSCP stapling constant](https://github.com/aws/s2n-tls/blob/282312cfacb58c2f31a28b76deef65fce44f608d/api/s2n.h#L577) which remains as  [TLS_STATUS_EXTENSION_REQUEST](https://github.com/aws/s2n-tls/blob/282312cfacb58c2f31a28b76deef65fce44f608d/tls/s2n_tls_parameters.h#L92) in tls/s2n_tls_parameters.h.
Don't see any issues for that, so why this one?

I appreciate the deliberate decision-making. Feel free to close this PR if we're done here.